docs/examples/tunnel-with-reverse-proxy/README.md

```diff
@@ -0,0 +1,161 @@
+# Bring your own reverse proxy
+
+> This builds on [the getting started guide](../../getting-started.md), and it is recommended to read that first.
+
+## Motivation
+
+Cloudflare tunnels can be used as a reverse proxy, or can forward traffic to an existing reverse proxy.
+This example shows how to configure cloudflared using the cloudflare-operator to front a reverse proxy. 
+In effect, we are using it as a layer 7 or layer 4 load balancer depending on whether we forward TCP/UDP or HTTP/HTTPS traffic.
+
+This configuration has a few use cases:
+- You have significant existing reverse proxy configuration, which you may or may not want to migrate to cloudflared.
+  This configuration allows additional routes to be added with a tunnel binding to use this reverse proxy in combination with other routes.
+- You require reverse proxy functionality that cloudflared does not have
+- You want to keep your reverse proxy configuration modular, so that you can use any load balancer implementation
+- you use a service mesh with its own ingress implementation
+
```

Comment:
```suggestion
- You have multiple load balancers pointing at your ingress. For example a lab might include metallb + DNS entries so that you are not traversing the internet when on the same network as your cluster
```